src/rocketchat.coffee

```diff
@@ -153,12 +153,13 @@ class RocketChatBotAdapter extends Adapter
 
 					if curts > @lastts
 						@lastts = curts
+						
+						user = @robot.brain.userForId newmsg.u._id, name: newmsg.u.username
```

Comment:
@geekgonecrazy I can confirm this change fixes some issues we've had with hubot-auth. For instance, if you set an auth role by DM with hubot and then execute a script from a channel, you'll receive an error that the user does not have role x. I found that the roles that were persisted in the brain were lost when the user changed rooms. This solves the issue.